### DISCUSSION TITLE
Validation attribute `Exists` don't respect `where` parameter

### DISCUSSION THREAD

**Discussion by alexrififi** [spatie/laravel-data#133] [Q&A] [answered]
Thanks for this powerful package!

Example Data class
```php
use Spatie\LaravelData\Attributes\Validation\Exists;

class Test extends \Spatie\LaravelData\Data
{
    public static function rules()
    {
        return [
            'location_id' => [
                new Exists(
                    AnyModel::class,
                    'id',
                    where: function ($query) {
                        // any condition
                        return $query->where('project_id', 1);
                    }
                ),
            ],
        ];
    }
}
```

## What's happening

Parameter `where` will not be used during validation, because Data's rule is converted to a string [when the rule is preparing](https://github.com/laravel/framework/blob/bc4f436b91ef1fae8884ed517f1c3abb6f4d2bb6/src/Illuminate/Validation/ValidationRuleParser.php#L111), but **native rule is not converted**!

Method `prepareRule` must return native instance of `Illuminate\Validation\Rules\Exists`, but **not string**!

When returning a string, the where condition is lost. The [documentation clearly shows](https://laravel.com/docs/9.x/validation#specifying-a-custom-column-name) that the line is not suitable for such a case

Here is [the condition](https://github.com/laravel/framework/blob/bc4f436b91ef1fae8884ed517f1c3abb6f4d2bb6/src/Illuminate/Validation/ValidationRuleParser.php#L117-L122)
```php
        if (! is_object($rule) ||
            $rule instanceof RuleContract ||
            ($rule instanceof Exists && $rule->queryCallbacks()) ||
            ($rule instanceof Unique && $rule->queryCallbacks())) {
            return $rule;
        }
```

I still don't understand how to fix it, but I think it needs to be fixed.
If you direct me on the right path, of course I will create a PR

----

For now, I'm using the native rule

```php
use Illuminate\Validation\Rule;

class Test extends \Spatie\LaravelData\Data
{
    public static function rules()
    {
        return [
            'location_id' => [
                Rule::exists(AnyModel::class, 'id')->where(fn($query) => $query->where('project_id', 1))
            ],
        ];
    }
}
```

**Comment by rubenvanassche** [comment]
Normally this should be fixed in v2 since we won't try to convert rules into strings anymore.

**Comment by alexrififi** [comment]
It still doesn't work in v2.

[Laravel converts the rule to a string](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Validation/ValidationRuleParser.php#L135) and the where condition is lost

**Comment by rubenvanassche** [comment, marked as answer]
This should be fixed with the next release (81bae97355e1f6fdfea3499b50649bcfd96b1d22)
